lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -340,22 +346,53 @@ private static Domain getDomain(
-                // PARQUET-1065 deprecated them. The result is that any writer that produced stats was producing unusable incorrect values, except
-                // the special case where min == max and an incorrect ordering would not be material to the result. PARQUET-1026 made binary stats
-                // available and valid in that special case
-                if (!(min instanceof Binary) || !(max instanceof Binary) || !min.equals(max)) {
+            if (column.getPrimitiveType().getPrimitiveTypeName().equals(INT96)) {
+                TrinoTimestampEncoder<?> timestampEncoder = createTimestampEncoder((TimestampType) type, timeZone);
+                List<Object> values = new ArrayList<>();
+                for (int i = 0; i < minimums.size(); i++) {
+                    Object min = minimums.get(i);
+                    Object max = maximums.get(i);
+
+                    // Parquet INT96 timestamp values were compared incorrectly for the purposes of producing statistics by older parquet writers, so
+                    // PARQUET-1065 deprecated them. The result is that any writer that produced stats was producing unusable incorrect values, except
+                    // the special case where min == max and an incorrect ordering would not be material to the result. PARQUET-1026 made binary stats
+                    // available and valid in that special case
+                    if (!(min instanceof Binary) || !(max instanceof Binary) || !min.equals(max)) {
+                        return Domain.create(ValueSet.all(type), hasNullValue);
+                    }
+
+                    values.add(timestampEncoder.getTimestamp(decode((Binary) min)));
+                }
+                return Domain.multipleValues(type, values, hasNullValue);
+            }
+            else if (column.getPrimitiveType().getPrimitiveTypeName().equals(INT64)) {
+                LogicalTypeAnnotation logicalTypeAnnotation = column.getPrimitiveType().getLogicalTypeAnnotation();
+                if (!(logicalTypeAnnotation instanceof TimestampLogicalTypeAnnotation)) {
+                    // Invalid statistics. Unit and UTC adjustment are not known
                     return Domain.create(ValueSet.all(type), hasNullValue);
                 }
 
-                values.add(createTimestampEncoder((TimestampType) type, timeZone).getTimestamp(decode((Binary) min)));
+                TimestampLogicalTypeAnnotation timestampTypeAnnotation = (TimestampLogicalTypeAnnotation) logicalTypeAnnotation;
+                if (timestampTypeAnnotation.getUnit() == null) {
+                    return Domain.create(ValueSet.all(type), hasNullValue);
+                }
+                // TODO: TimeZone should probably be based on timestampTypeEncoding.isAdjustedToUTC()
+                TrinoTimestampEncoder<?> timestampEncoder = createTimestampEncoder((TimestampType) type, DateTimeZone.UTC);
```

Comment:
as we talked, this should match the colum reader, which for in64 timestamps currently  does not do any adjustments. Seems this should only be done when `isAdjustedToUTC=false`, 
so here we can bail out when `isAdjustedToUTC=true` (with a TODO comment)